test/Common/DurableOrchestrationClientBaseTests.cs

```diff
@@ -23,5 +27,103 @@ public async Task StartNewAsync_is_calling_overload_method()
             var result = await durableOrchestrationClientBaseMock.Object.StartNewAsync(functionName, null);
             result.Should().Be(instanceId);
         }
+
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task RaiseEventAsync_InvalidInstanceId_ThrowsException()
+        {
+            var instanceId = Guid.NewGuid().ToString();
+            var orchestrationServiceClientMock = new Mock<IOrchestrationServiceClient>();
+            orchestrationServiceClientMock.Setup(x => x.GetOrchestrationStateAsync(It.IsAny<string>(), It.IsAny<bool>())).ReturnsAsync(GetInvalidInstanceState());
+            var durableOrchestrationClient = new DurableOrchestrationClient(orchestrationServiceClientMock.Object, GetDurableTaskExtension(), new OrchestrationClientAttribute { });
+
+            await Assert.ThrowsAnyAsync<ArgumentException>(async () => await durableOrchestrationClient.RaiseEventAsync("invalid_instance_id", "anyEvent", new { message = "any message" }));
```

Comment:
I don't love adding a dependency on what type of exception we are throwing in the test, as it makes the test more fragile if we change the exception type in the future.

That being said, it seems preferable to allowing any exception, as if future code changes cause the test to break (like `RaiseEventAsync` encountering a `NullReferenceException`), the test would still pass if we allowed any exception.

Reply:
I think it's better to be specific. If we change the type of exception we throw, that could be considered a breaking change and we'd want our tests to catch this.